libbeat/testing/integration/run_beat.go

```diff
@@ -73,8 +73,14 @@ func (b *RunningBeat) CollectOutput(limit int) string {
 		if err != nil {
 			builder.WriteString(l)
 		} else {
-			pretty, _ := json.MarshalIndent(m, "", "  ")
-			builder.Write(pretty)
+			var data []byte
+			if pretty {
+				data, _ = json.MarshalIndent(m, "", "  ")
+			} else {
+				data, _ = json.Marshal(m)
+			}
+
+			builder.Write(data)
```

Comment:
I think we should not even `Unmarshal` on line 72 if `pretty = false`.

It called `Unmarshal` only to check if that's valid JSON and we can reformat it. Now when `pretty` is optional, we can skip this step if we don't care to reformat anyway.